File: pset6/sentimental-readability/readability.py
# TODO
import cs50
import re

text = cs50.get_string("Text: ")


def main():
    # number of letters
    letters = count_letters(text)
    # number of words
    words = count_words(text)
    # number of  sentences
    sentences = count_sentences(text)
    print(sentences)

    # calculate the average of letters and sentences
    lavg = letters / words * 100
    savg = sentences / words * 100
    # apply the Coleman-Liau index
    grade = 0.0588 * lavg - 0.296 * savg - 15.8
    # rounds the grade so it is an int
    gradeint = round(grade)

    # print grades
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print("Grade", gradeint)

# Counts how many letters in text using a for loop


def count_letters(text):
    return len([x for x in text if x.isalpha()])

# counts the number of words using the strip/split functions


def count_words(text):
    return (len(text.strip().split(" ")))

# counts the number of sentences using regex and split function


def count_sentences(text):
    sentence = (re.split(r'[.|!|?]', text))
    return(len(sentence) - 1)


main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)

{
    //get input from user
    string text = get_string("Text: ");

    //number of letters
    float letters = count_letters(text);
    //number of words
    float words = count_words(text);
    //number of sentences
    float sentences = count_sentences(text);

    //print the number of letters, words and sentences
    //printf("%f letter(s)\n", letters);
    //printf("%f word(s) \n", words);
    //printf("%f sentence(s) \n", sentences);
    
    //Calculate the average number of letters per 100 words
    float lavg = letters / words * 100;
    //Calculate the average number of sentences per 100 words
    float savg = sentences / words * 100;
    //Calculates the Coleman-Liau Index
    float grade = 0.0588 * lavg - 0.296 * savg - 15.8;
    
    //Cast float grade to int gradeint
    int gradeint = round(grade);
    
    //Print the results of the Index
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", gradeint);
    }
    

    
    //print the grade based on the Coleman-Liau index.
    //printf( "%f \n", lavg);
    //printf("%f \n", savg);
    //printf("grade: %f \n", round (grade));
}

int count_letters(string text)
{
    int countl = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalnum(text[i]))
        {
            countl++;
        }
    }
    return countl;
}
int count_words(string text)
{
    int countw = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            countw++;
        }
    }
    countw = countw + 1;
    return countw;
}
int count_sentences(string text)
{
    int counts = 0;
    char p1 = 33;
    char p2 = 46;
    char p3 = 63;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == p1 || text[i] == p2 || text[i] == p3)
        {
            counts++;
        }
    }
    return counts;
}




